src/runtime/proxy.ts

```diff
@@ -1,17 +1,116 @@
 import { Context } from "../runtime/mod.ts";
 import { RequestOf, ResponseOf } from "../types/registry.ts";
 
+/**
+ * This type is a union of all of the valid script names for a SPECIFIC module
+ * in a registry.
+ */
 type AnyScriptName<RegistryT, Key extends keyof RegistryT> = keyof RegistryT[Key];
+
+/**
+ * This type is a union of ALL valid [<module>, <script-in-module>] pairs in a
+ * registry.
+ * 
+ * This is used primarily in {@link MapFrom} to denote a valid mapping into a
+ * registry.
+ */
```

Comment:
I dont understand the `{ ... }[keyof ...]` syntax, could an example be provided here?